low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -249,18 +249,35 @@ class ResetCmd(Cmd):
 
     name = "reset"
     script = "sspl_reset"
+    process_class=None
 
     def __init__(self, args):
         super().__init__(args)
 
     def validate(self):
-        # Common validator classes to check Cortx/system wide validator
-        pass
+        if not self.args:
+            raise SetupError(1,
+                             "%s - validation failure. %s",
+                             self.name,
+                             "SSPL Reset requires the type of reset(hard|soft).")
+        i=0
+        while i < len(self.args):
+            try:
+                if self.args[i] == "hard":
+                    self.process_class = "HardReset"
+                elif self.args[i] == "soft":
+                    self.process_class = "SoftReset"
+            except (IndexError, ValueError):
+                raise SetupError(errno.EINVAL, "Invalid Argument for %s"% self.name)
+            i+=1
```

Comment:
Same comment as in.
https://github.com/Seagate/cortx-monitor/pull/275#discussion_r564937963

Reply:
Done.